src/main/java/org/gridsuite/modification/server/modifications/NetworkStoreListener.java

```diff
@@ -130,9 +119,68 @@ public void onElementReplaced(Identifiable identifiable, String attribute, Objec
         addSimpleModificationImpact(identifiable);
     }
 
+    @Override
+    public void onUpdate(Identifiable identifiable, String attribute, Object oldValue, Object newValue) {
+        networkImpacts.add(
+            SimpleElementImpact.builder()
+                .impactType(SimpleElementImpact.SimpleImpactType.MODIFICATION)
+                .elementType(identifiable.getType())
+                .elementId(identifiable.getId())
+                .substationIds(getSubstationIds(identifiable))
+                .build()
+        );
+        updateEquipmentIndexation(identifiable, attribute, networkUuid, network.getVariantManager().getWorkingVariantId());
+    }
+
     @Override
     public void onUpdate(Identifiable identifiable, String attribute, String variantId, Object oldValue, Object newValue) {
         addSimpleModificationImpact(identifiable);
+        updateEquipmentIndexation(identifiable, attribute, networkUuid, network.getVariantManager().getWorkingVariantId());
+    }
+
+    private void updateEquipmentIndexation(Identifiable<?> identifiable, String attribute, UUID networkUuid, String variantId) {
+        equipmentInfosService.updateEquipment(identifiable, networkUuid, variantId);
+        // because all each equipment carry its linked voltage levels/substations name within its document
+        // if attribute is "name" and identifiable type is VOLTAGE_LEVEL or SUBSTATION, we need to update all equipments linked to it
+        if (attribute.equals("name") && (identifiable.getType().equals(IdentifiableType.VOLTAGE_LEVEL) || identifiable.getType().equals(IdentifiableType.SUBSTATION))) {
+            updateLinkedEquipments(identifiable);
+        }
+    }
+
+    private void updateLinkedEquipments(Identifiable<?> identifiable) {
+        if (identifiable.getType().equals(IdentifiableType.VOLTAGE_LEVEL)) {
+            VoltageLevel updatedVoltageLevel = network.getVoltageLevel(identifiable.getId());
+            // update all equipments linked to voltageLevel
+            updateEquipmentsLinkedToVoltageLevel(updatedVoltageLevel);
+            // update substation linked to voltageLevel
+            Optional<Substation> linkedSubstation = updatedVoltageLevel.getSubstation();
+            if (linkedSubstation.isPresent()) {
+                createdEquipments.add(EquipmentInfos.toInfosWithUpdatedVoltageLevelName(linkedSubstation.get(), updatedVoltageLevel, networkUuid, network.getVariantManager().getWorkingVariantId()));
```

Comment:
If I understood well, maybe rename `createdEquipments` into `createdOrUpdatedEquipments` ? Or use a new list `updatedEquipments` ?
It's consufing to me, it took some time to understand why do you do `add` on a list named `createdEquipments` (I don't know this class and no Javadoc so not easy to get in)

Reply:
Yes use `modifedEquipments` list ?

Reply:
fixed